src/smile_parser.js

```diff
@@ -1,39 +1,79 @@
-const emojis = require('./minifiedEmoji.json')
+const emojis = require("./minifiedEmoji.json")
 const regex = new RegExp(/:([a-zA-Z1-9+_-]*?):/)
 const toneRegex = new RegExp(/:(.*?)::(skin-tone-\d*):/)
-const emojiRegex = new RegExp(/(\B|^)((:\)|:\/|:\(|:'\(|:\||;\))\B|(:D|:P|:o)\b)/)
-let emoji
+const emojiRegex = new RegExp(
+  /(\B|^)((:\)|:\/|:\(|:'\(|:\||;\))\B|(:D|:P|:o)\b)/
+)
+
+const emojiMap = {}
+
+for (const emoji of emojis) {
+  if (emoji.short_names) {
+    for (const short_name of emoji.short_names) {
+      emojiMap[`:${short_name}:`] = emoji
+    }
+  }
+
+  if (emoji.texts) {
+    for (const text of emoji.texts) {
+      emojiMap[text] = emoji
+    }
+  }
+}
```

Comment:
Emojis are added to the object to possibly multiple keys if they have multiple short names and/or texts.

`emojiMap[":)"]` or `emojiMap[":grin:"]` would return an emoji.